pkg/manager/member/scaler.go

```diff
@@ -53,29 +55,45 @@ type generalScaler struct {
 	pvcControl controller.PVCControlInterface
 }
 
-func (gs *generalScaler) deleteDeferDeletingPVC(tc *v1alpha1.TidbCluster,
+func (gs *generalScaler) deleteDeferDeletingPVC(controller runtime.Object,
 	setName string, memberType v1alpha1.MemberType, ordinal int32) (map[string]string, error) {
-	ns := tc.GetNamespace()
+	meta := controller.(metav1.Object)
+	ns := meta.GetNamespace()
 	// for unit test
 	skipReason := map[string]string{}
-
-	// pvcName := ordinalPVCName(memberType, setName, ordinal)
-	podName := ordinalPodName(memberType, tc.Name, ordinal)
-	l := label.New().Instance(tc.GetInstanceName())
-	l[label.AnnPodNameKey] = podName
+	var podName, kind string
+	var l label.Label
+	_, ok := controller.(*v1alpha1.TidbCluster)
```

Comment:
maybe it's better to use type switches here, example: https://tour.golang.org/methods/16